core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1224,7 +1227,8 @@ private[spark] class DAGScheduler(
       logInfo(s"Submitting ${tasks.size} missing tasks from $stage (${stage.rdd}) (first 15 " +
         s"tasks are for partitions ${tasks.take(15).map(_.partitionId)})")
       taskScheduler.submitTasks(new TaskSet(
-        tasks.toArray, stage.id, stage.latestInfo.attemptNumber, jobId, properties))
+        tasks.toArray, stage.id, stage.latestInfo.attemptNumber, jobId, properties,
+        stage.latestInfo.attemptNumber > 0 && stage.isIndeterminate))
```

Comment:
Here we can't simply use `stage.isIndeterminate`, because isIndeterminate will call rdd.outputDeterministicLevel, which is a lazy val in RDD. If we trigger outputDeterministicLevel in the first attempt, `doCheckpoint` will never take effect, it will break [the UT](https://github.com/apache/spark/blob/e4a968d82947a68d2daedc8e5fb9adf57cf9bd5b/core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala#L2787) in DAGSchedulerSuite.